src/pipx/commands/run.py

```diff
@@ -348,7 +361,8 @@ def _get_requirements_from_script(content: str) -> Optional[List[str]]:
     pyproject = tomllib.loads(content)
 
     requirements = []
-    for requirement in pyproject.get("run", {}).get("requirements", []):
+    req_table = pyproject if script else pyproject.get("run", {})
+    for requirement in req_table.get("requirements", []):
```

Comment:
isn't the table called `dependencies`?
```suggestion
    for requirement in req_table.get("dependencies", []):
```

Reply:
Oh, wow. That was an issue before too. Okay, I think since the implementation was wrong we can go ahead and make it a hard (but nicely descriptive) error if the old form is encountered?